app/graphql/resolvers/base_resolver.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class BaseResolver
+  attr_accessor :error
+
+  def initialize(context:, arguments:, object:)
+    @context = context
+    @arguments = arguments
+    @object = object
+    @error = nil
+  end
+
+  def valid?
+    raise 'Implement in subclass'
```

Comment:
No need for action here, just a question about ruby types. In C#, we have the concept of an abstract base class, specifically to avoid things like throwing exceptions from a base class when the subclass must implement. Does such a thing exist in ruby?

Reply:
Nope what you see here is pretty common. 🤷‍♂